rules/mapping.go

```diff
@@ -71,6 +72,22 @@ func newMappingRuleSnapshot(
 	}, nil
 }
 
+func newMappingRuleSnapshotFromFields(
+	name string,
+	tombstoned bool,
+	cutoverNanos int64,
+	tagFilters map[string]string,
+	policies []policy.Policy,
+) mappingRuleSnapshot {
+	return mappingRuleSnapshot{
```

Comment:
Do you need to initialize the `filter` field here? Also perhaps return a `*mappingRuleSnapshot` here to consistent w/ the other ctor?